examples/packet_mirroring/main.tf

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+terraform {
+  required_providers {
+    google = {
+      version = ">= 3.45.0"
+    }
+    null = {
+      version = ">= 2.1.0"
+    }
+  }
+}
+
+resource "google_compute_network" "default" {
```

Comment:
Please use the module for creating the network + subnets.

Reply:
I tried to do this, but I couldn't figure out how to refer to the module's subnet from the resources where the subnet's self-link is required. I kept getting errors such as this:

```
Error: Unsupported attribute

  on main.tf line 57, in resource "google_compute_instance" "mirror":
  57:     subnetwork = module.test-vpc-module.subnets.subnet-name[0].self_link
    |----------------
    | module.test-vpc-module.subnets is object with 1 attribute "us-central1/my-subnet"

This object does not have an attribute named "subnet-name".
```

So I finally gave up and made the network and subnet into variables.

Reply:
Alright, that's fine.